model-optimizer/extensions/middle/dequantize_linear_resolver.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from extensions.ops.Cast import Cast
+from extensions.ops.elementwise import Mul, Sub
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph
+from mo.middle.passes.convert_data_type import data_type_str_to_np
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.reshape import Reshape
+
+
+class DequantizeLinearResolver(MiddleReplacementPattern):
+    """
+    DequantizeLinear can be replace with the following formula: y = (x - x_zero_point) * x_scale
+    """
+    enabled = True
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('dequantize', dict(op='DequantizeLinear')),
+            ],
+            edges=[]
+        )
+
+    def replace_pattern(self, graph: Graph, match: dict):
+        dequantize_node = match['dequantize']
+        node_name = dequantize_node.soft_get('name', dequantize_node.id)
+        axis = dequantize_node.soft_get('axis', None)
+        scale_y_shape = dequantize_node.in_port(1).get_source().node.shape
+        model_data_type = data_type_str_to_np(graph.graph['cmd_params'].data_type)
+        cast = Cast(graph, {'dst_type': model_data_type, 'name': node_name + '/Cast'}).create_node()
+        dequantize_node.in_port(0).get_connection().set_destination(cast.in_port(0))
+        mul = Mul(graph, {}).create_node()
+
+        is_second_port_connected = dequantize_node.is_in_port_connected(2)
+        if is_second_port_connected:
+            sub = Sub(graph, {'name': node_name + '/Sub'}).create_node()
+            cast.out_port(0).connect(sub.in_port(0))
+            dequantize_node.in_port(2).get_connection().set_destination(sub.in_port(1))
+            sub.out_port(0).connect(mul.in_port(0))
+        else:
+            cast.out_port(0).connect(mul.in_port(0))
+
+        dequantize_node.in_port(1).get_connection().set_destination(mul.in_port(1))
+        dequantize_node.out_port(0).get_connection().set_source(mul.out_port(0))
+
+        if axis and len(scale_y_shape) > 0 and scale_y_shape[0] > 1:
```

Comment:
`if axis` is not a valid check, because it `axis = 0` then the result will be false and you will not go inside the `if`. The valid check is `if axis is not None`